zk_shell/tests/test_keys.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+""" keys test cases """
+
+import unittest
+
+from zk_shell.keys import Keys
+
+
+class KeysTestCase(unittest.TestCase):
+    """ test keys """
+
+    def setUp(self):
+        """
+        nothing for now
+        """
+        pass
+
+    def test_extract(self):
```

Comment:
Maybe add a case for `Missing`?

Reply:
@Yasumoto: hmm, don't think it's needed because you _must_ call `validate_one()` before calling `extract()`. Hmm, maybe it should be private (e.g.: `_extract()`).